### PR TITLE
fixed dataset csv that crashes after ~4500 datasets. #288

### PR DESCRIPTION
**Pull request by vinyll**

### CODE REVIEW

**Comment by davidbgk** [comment]
Might be worth a test?

**Comment by noirbizarre** [comment]
+1 for the test ;)

**Comment by vinyll** [comment]
The model GeoZone has changed to set_name_ as _required_. Therefore we cannot reproduce the same context where a GeoZone has no name.
Should we run a migration and export the _id_ for the default _name_?

**Comment by vinyll** [comment]
Seen with @davidbgk:
The GeoZone should not have empty _name_. The GeoZone json file dump have all _features_ with a _name_ property.
The `load()` task does not drop the collection (see https://github.com/etalab/udata/blob/master/udata/core/spatial/commands.py#L29).
This may result into older zones that do not have any _name_ field now.

**Comment by noirbizarre** [comment]
OK for me, this is fixed by the new GeoZones enforced specifications. :)

**Comment by vinyll** [comment]
indeed it is. Therefore I wonder if we still need this as it is due to old existing data in the data.

**Comment by vinyll** [comment]
We should reset the geozones in prod to check if we still have the issue.
If not we should ignore this PR.

**Comment by davidbgk** [comment]
Let's reset for the territory deployment, we'll need it anyway :)

File: udata/tests/dataset/test_csv.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class DatasetModelTest(TestCase, DBTestMixin):
+    def test_owned_by_user(self):
+        user = UserFactory()
+        datasets = [
+            DatasetFactory(owner=user, spatial=SpatialFactory(
+                zones=[GeoZoneFactory(name='')])),
+            DatasetFactory(owner=user, spatial=SpatialFactory()),
+        ]
+
+        self.assertEqual(len(result), 1)
```
Comment (davidbgk) [inline udata/tests/dataset/test_csv.py]: Unfinished test?

**Comment by vinyll** [comment]
we may leave this unmerged and delete. GeoZone have changed and this bug should not happen with the new GeoZone model.
Clearing out outdated geozone data and rebuilding should fix.

**Comment by vinyll** [comment]
After discussing we'll drop this resolution and go for skipping a line that fails, whatever the error.

**Comment by davidbgk** [comment]
Should be fixed by https://github.com/etalab/udata/pull/363

**Comment by vinyll** [comment]
Yup. I close this

**Pull request closed without merging**